utils/guardrails/guard.py

```diff
@@ -0,0 +1,204 @@
+                **params,
+                model_kwargs={
+                "select_expert": "Meta-Llama-Guard-2-8B",
+                "process_prompt": False,
+                "do_sample": False,
+                "max_tokens_to_generate": 1024,
+                "temperature": 0.1,
+                },
+            )
+        elif api == "sambaverse":
+            llm = Sambaverse(
+                **params,
+                sambaverse_model_name="Meta/Meta-Llama-Guard-2-8B",
+                model_kwargs={
+                    "select_expert": "Meta-Llama-Guard-2-8B",
+                    "process_prompt": True,
+                    "do_sample": False,
+                    "max_tokens_to_generate": 1024,
+                    "temperature": 0.1,
+                },
+            )
+        else:
+            raise ValueError(f"Invalid LLM API: {api}, only'sambastudio' and'sambaverse' are supported.")
+        return llm
+
+    def evaluate(
+        self,
+        input: Union[List[Dict], str],
+        role: str,
+        error_message: str = None,
+        return_guardrail_type: bool = True,
+        raise_exception: bool =False
+        ) -> Union[str, List[str]]: 
+        """
+        Evaluate a message or a conversation against the guardrails.
+
+        Parameters:
+        - input (Union[List[Dict], str]): The message or conversation to evaluate. It can be a string with the message or a list of dictionaries representing a conversation.
+            Example conversation input
+            [
+             {"message_id":0,"role":"user", "content":"this is an user message"},
+             {"message_id":1,"role":"assistant","content":"this is an assistant response"},
+            ]
+        - role (str): The role of the message to analyse. It can be either 'user' or 'assistant'.
+        - error_message (str, optional): The error message to be displayed when the message violates the guardrails. If not provided, a default message "The message violate guardrails" will be used.
+        - return_guardrail_type (bool, optional): If True, the function will return the violated guardrail types. If False, it will only return the error message. Default is True.
+        - raise_exception (bool, optional): If True, the function will raise a ValueError with the error message when the message violates the guardrails. If False, it will return the error message. Default is False.
+
+        Returns:
+        - Union[str, List[str]]: The result of the evaluation. If the message violates the guardrails, it will return the error message with the violated guardrail types based on the return_guardrail_type parameter.
+        If the message passes the guardrails, it will return the input message or conversation.
+
+        Raises:
+        - ValueError: If the role is not 'user' or 'assistant'.
+        - ValueError: If the message violates the guardrails when raises_exception is True.
+        """
+
+        if isinstance(input, str):
```

Comment:
Here and in other places it would be more readable if we added more blank lines between the different paragraphs / actions, sometimes preceded by general comments.

Reply:
added